### PR TITLE
Core/MySQL: wait for the DB to be ready

### LINKED ISSUE #1285: Authserver and Worldserver should wait for MySQL until it's ready
Currently our main processes `authserver` and `worldserver` just die if they cannot connect to the mysql server.

Ideally they should retry to connect to the database periodically until they either connect to it or timeout.

This will help all kind of virtual machine or docker or similar cloud-friendly setups.

### PR DESCRIPTION
**Pull request by FrancescoBorzi**
##### CHANGES PROPOSED:

Authserver and Worldserver don't die immediately if they don't find MySQL, but they wait until they can connect to it (trying every 10 seconds).

After 180 attempts (30 minutes) it exits.

This is how modern, resilient apps are done. This helps virtualization environments & similar setups.

image

image

###### ISSUES ADDRESSED:

-  Closes #1285

Closes 

##### TESTS PERFORMED:
- Compiles on Linux
- Successfully waits until mysql service is up and then connects to it
- Logging in game works

##### HOW TO TEST THE CHANGES:

- Stop mysql service (on linux: `sudo service mysql stop`)
- Run authserver and worldserver
- After a while, start mysql service again (on linux: `sudo service mysql start`)

On windows it should be the same, just a different way to start/stop mysql I guess.

##### Target branch(es):

Master

### CODE REVIEW

File: src/common/Database/MySQLConnection.cpp [hunk 1]
```diff
@@ -107,36 +112,59 @@ bool MySQLConnection::Open()
...
-    if (m_Mysql)
-    {
-        if (!m_reconnecting)
+    // Possible improvement for future: make ATTEMPTS and SECONDS configurable values
+    uint const ATTEMPTS = 180;
+    uint const SECONDS = 10;
+
+    uint count = 0;
+    do {
+        m_Mysql = mysql_real_connect(
```
Comment (Viste) [inline src/common/Database/MySQLConnection.cpp]: why you split it line by line in string more easy to read i think

Comment (FrancescoBorzi) [reply src/common/Database/MySQLConnection.cpp]: too-long lines are not good, better to split for readability :)

Comment (Viste) [reply src/common/Database/MySQLConnection.cpp]: ok no problem just ask

File: src/common/Database/MySQLConnection.cpp [hunk 2]
```diff
@@ -107,36 +112,59 @@ bool MySQLConnection::Open()
...
     #endif
 
-    m_Mysql = mysql_real_connect(mysqlInit, m_connectionInfo.host.c_str(), m_connectionInfo.user.c_str(),
-        m_connectionInfo.password.c_str(), m_connectionInfo.database.c_str(), port, unix_socket, 0);
-
-    if (m_Mysql)
-    {
-        if (!m_reconnecting)
+    // Possible improvement for future: make ATTEMPTS and SECONDS configurable values
+    uint const ATTEMPTS = 180;
```
Comment (Winfidonarleyan) [inline src/common/Database/MySQLConnection.cpp]: Please replace all types `uint` to `uint32`. In VS 2017 error: `'identifier' : undeclared identifier`

**Comment by Winfidonarleyan** [comment]
I tested in Win (after repalce types `uint`).
- authserver (not clean core)
image
- worldserver (not clean core)
image

So, worked perfectly.

**Comment by FrancescoBorzi** [comment]
Thanks a lot @Winfidonarleyan for the nice feedback!

File: src/common/Database/MySQLConnection.cpp (hunk 2 above)
Comment (FrancescoBorzi) [reply src/common/Database/MySQLConnection.cpp]: fixed, thanks a lot!

**Comment by FrancescoBorzi** [comment]
@BarbzYHOOL also be aware that this code is about the bootstrapping phase, I'm not sure if it can solve that specific problem that you described.

**Comment by BarbzYHOOL** [comment]
https://github.com/azerothcore/azerothcore-wotlk/commit/e5ad0165b36c62bb90fabea242427ba08f20695f

https://github.com/azerothcore/azerothcore-wotlk/commit/27ea9c3f9558168d182f3dc28401e2d341a9ed81

@FrancescoBorzi twice the same commit no?

**Comment by FrancescoBorzi** [comment]
@BarbzYHOOL weird, I was sure to have squashed lol (normally I do that).

So either I'm going crazy or this other merge caused the confusion https://github.com/azerothcore/azerothcore-wotlk/commit/d5294f7112798f1bf2053b5fb5e8616483981ea1

**Comment by poszer** [comment]
Tested on windows, working great!
screenshot_3

To test changes on windows : 
1. Start Command Prompt ( You **MUST** Run as Administrator )
2. Type net stop MySQLxx   ( xx = Your version of MySQL, in my case will be MySQL56 )
3. Run Auth & Worldserver 
4. To start MySQL again run command  net start MySQLxx ( xx = Your MySQL version )

**Pull request merged**